send-transaction-service/src/send_transaction_service.rs

```diff
@@ -602,7 +628,8 @@ impl SendTransactionService {
     ) -> ProcessTransactionsResult {
         let mut result = ProcessTransactionsResult::default();
 
-        let mut batched_transactions = HashSet::new();
+        let mut batched_transactions = Vec::new();
```

Comment:
I think hashset was used to have unique transaction, could you clarify why it is not needed? Are they already filtered here by construction?

Reply:
transactions already unique because filled from `transactions` object that is `HashMap`

Reply:
good catch